docs/framework/interop/default-marshalling-for-strings.md

```diff
@@ -242,49 +239,56 @@ int GetWindowText(
 );
 ```
 
-A `StringBuilder` can be dereferenced and modified by the callee, provided it does not exceed the capacity of the `StringBuilder`. The following code example demonstrates how `StringBuilder` can be initialized to a fixed length.
+A `char[]` can be dereferenced and modified by the callee. The following code example demonstrates how `ArrayPool<char>` can be used to pre-allocate a `char[]`.
 
 ```csharp
 using System;
+using System.Buffers;
 using System.Runtime.InteropServices;
-using System.Text;
 
 internal static class NativeMethods
 {
-    [DllImport("User32.dll")]
-    internal static extern void GetWindowText(IntPtr hWnd, StringBuilder lpString, int nMaxCount);
+    [DllImport("User32.dll", CharSet = CharSet.Unicode)]
+    public static extern void GetWindowText(IntPtr hWnd, [Out] char[] lpString, int nMaxCount);
 }
 
 public class Window
 {
     internal IntPtr h;        // Internal handle to Window.
-    public String GetText()
+    public string GetText()
     {
-        StringBuilder sb = new StringBuilder(256);
-        NativeMethods.GetWindowText(h, sb, sb.Capacity + 1);
-        return sb.ToString();
+        char[] buffer = ArrayPool<char>.Shared.Rent(256 + 1);
+        NativeMethods.GetWindowText(h, buffer, buffer.Length);
+        return new string(buffer);
```

Comment:
Should this be as follows?

```csharp
var textLength = NativeMethods.GetWindowText(h, buffer, buffer.Length);
return new string(buffer, 0, textLength);
```

Otherwise, the returned string may have literal `\0` characters at the end, since its length will be `buffer.Length`.

Reply:
Also, we should return the shared buffer:

```csharp
var windowText = new string(buffer, 0, textLength);
ArrayPool<char>.Shared.Return(buffer);
return windowText;
```

Reply:
And, while we are at it, this article is in .NET Framework section, but sample uses ArrayPool, which requires additional System.Buffers package dependency on it. Is it something we are concerned about? I'd personally not add extra dependency in my .NET Framework apps just for P/Invoke, even if it makes it faster.